### PR TITLE
created admin/speech controller and implemented methods

### PR DESCRIPTION
**Pull request by davidmetta**

### CODE REVIEW

File: app/controllers/admin/speeches_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  def new_speech
+    @speech = Speech.new
+  end
+
+  def create_speech
+    @speech = Speech.new(speech_params)
+    if @speech.save
+      redirect_to admin_speech(@admin)
+      flash[:notice] = 'Nice! speech successfully created!'
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb:19]: line 19 before line 18

File: app/controllers/admin/speeches_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    end
+  end
+
+  def edit_speech
+  end
+
+  def update_speech
+    if @speech.update(speech_params)
+      redirect_to admin_speech(@speech)
+      flash[:notice] = 'Nice! speech successfully updated!'
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb:32]: line 32 before 31

File: app/controllers/admin/speeches_controller.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def new_speech
+    @speech = Speech.new
+  end
+
+  def create_speech
+    @speech = Speech.new(speech_params)
+    if @speech.save
+      redirect_to admin_speech(@admin)
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb]: @speech instead of @admin

File: app/controllers/admin/speeches_controller.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    else
+      flash[:notice] = 'oops! something went wrong, speech could not be updated!'
+      render :edit
+    end
+  end
+
+  def destroy_speech
+    if @speech.destroy
+      redirect_to admin_speeches
+      flash[:notice] = 'Nice! speech successfully deleted!'
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb:42]: line 42 before 41

File: app/controllers/admin/speeches_controller.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    @speech = Speech.new
+  end
+
+  def create_speech
+    @speech = Speech.new(speech_params)
+    if @speech.save
+      redirect_to admin_speech(@admin)
+      flash[:notice] = 'Nice! speech successfully created!'
+    else
+      flash[:notice] = 'oops! something went wrong, speech could not be created!'
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb]: flash[:alert]

File: app/controllers/admin/speeches_controller.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  def edit_speech
+  end
+
+  def update_speech
+    if @speech.update(speech_params)
+      redirect_to admin_speech(@speech)
+      flash[:notice] = 'Nice! speech successfully updated!'
+    else
+      flash[:notice] = 'oops! something went wrong, speech could not be updated!'
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb]: flash[:alert]

File: app/controllers/admin/speeches_controller.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      render :edit
+    end
+  end
+
+  def destroy_speech
+    if @speech.destroy
+      redirect_to admin_speeches
+      flash[:notice] = 'Nice! speech successfully deleted!'
+    else
+      flash[:notice] = 'oops! something went wrong, speech could not be deleted!'
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb]: flash[:alert]

File: app/controllers/admin/speeches_controller.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  private
+
+  def find_speech
+    @speech = Speech.find(params[:speech_id])
+  end
+
+  def speech_params
+    params.require(:speech).permit(:title, :date, :country, :city, :content, :duration)
```
Comment (ghost) [inline app/controllers/admin/speeches_controller.rb]: What about speaker and category? Must be able to change that as well

**Comment by arosty** [comment]
disagree

**Comment by ghost** [comment]
Fine for me

**Comment by lepetitbelge** [comment]
good for me

**Pull request merged**